### PR TITLE
#14 merge with issue branch 

### PR DESCRIPTION
**Pull request by ballaln**

### CODE REVIEW

File: lasso/dyna/ArrayType.py [hunk 1]
```diff
@@ -40,12 +40,15 @@ class ArrayType:
     element_solid_effective_plastic_strain = (
         "element_solid_effective_plastic_strain"  #: shape: (n_states, n_solid_layers, n_solids)
     )
-    element_solid_history_variables = "element_solid_history_variables"  #: shape: (n_states, n_solids, n_solid_layers, n_solids_history_vars)
+    #: shape: (n_states, n_solids, n_solid_layers, n_solids_history_vars)
+    element_solid_history_variables = "element_solid_history_variables"
```
Comment (codie3611) [inline lasso/dyna/ArrayType.py:44]: I haven't gotten to the docs yet but if you put it above then sphinx won't display the helptext. There might be other ways around this though. I would prefer anyway to move to mkdocs.

File: lasso/utils/rich_progress_bars.py [hunk 2]
```diff
@@ -67,6 +70,7 @@ def render(self, task: Any) -> str:
 
 
 class SubsamplingWaitTime(ProgressColumn):
+    """ *add description? """
```
Comment (codie3611) [inline lasso/utils/rich_progress_bars.py:73]: Simply add `TODO` next time as it is a universal flag easily found in any codebase.

File: lasso/utils/language.py [hunk 3]
```diff
@@ -19,16 +19,14 @@ def get_var(name, context, default=None):
...
-                return default
+            return default() if callable(default) else default
 
     return current_context
 
 
 def set_var(name, value, context):
-    """Set a viarble in a dict context
+    """
+    Set a viarble in a dict context
```
Comment (codie3611) [inline lasso/utils/language.py:29]:
```suggestion
    Set a variable in a dict context
```

**Review by codie3611 — COMMENTED** [review]
Hey, thanks for the PR :pray:  Two things, first please remove `.DS_Store` and add it to the `.gitignore` since you uploaded this mac-specific file twice to the repo (you can do another PR). Second please mark someone as a reviewer to ensure that the code was checked at least once. Thanks :+1:

File: lasso/dyna/ArrayType.py (hunk 1 above)
Comment (ballaln) [reply lasso/dyna/ArrayType.py:44]: I will check this out and change accordingly

File: lasso/utils/rich_progress_bars.py (hunk 2 above)
Comment (ballaln) [reply lasso/utils/rich_progress_bars.py:73]: I am quite new working with public repos. with adding TODO, do you mean writing the keyword inside the quotes or is it some kind of tag?

Comment (codie3611) [reply lasso/utils/rich_progress_bars.py:73]: Yeah writing simply the keyword. Otherwise we can also add a linting excuse if it does not look important.

**Pull request merged**